specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/examples/ManagedIdentity/TasksCreate_WithSystemAndUserIdentities.json

```diff
@@ -0,0 +1,247 @@
+{
+  "parameters": {
+    "subscriptionId": "4385cf00-2d3a-425a-832f-f4285b1c9dce",
+    "resourceGroupName": "myResourceGroup",
+    "registryName": "myRegistry",
+    "api-version": "2019-06-01-preview",
+    "taskName": "mytTask",
+    "taskCreateParameters": {
+      "properties": {
+        "status": "Enabled",
+        "platform": {
+          "os": "Linux",
+          "architecture": "amd64"
+        },
+        "agentConfiguration": {
+          "cpu": 2
+        },
+        "step": {
+          "imageNames": [
+            "azurerest:testtag"
+          ],
+          "dockerFilePath": "src/DockerFile",
+          "contextPath": "src",
+          "isPushEnabled": true,
+          "noCache": false,
```

Comment:
The model validator is right again. You did not specify the `"type"` of `"step"`. From the example it seems that the `"type": "Docker"` should be added to `"step"` for the validator to validate against `"DockerBuildStep"` as opposed to `"TaskStepProperties"`.

General comment: Please remove "type" from the required list of properties from the child model definition `"DockerBuildStep"` and add it to `"TaskStepProperties"`. Please make sure you do this for all the Polymorphic models.

Reply:
@shahzzzam - Please go through each and every error reported by the model validator and fix those issues.